### PR TITLE
Mat 4489 - Add Stratifications 

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR Template

Jira Ticket: [MAT-4489](https://jira.cms.gov/browse/MAT-4489)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e. Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: .github/workflows/maven-ci.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     permissions:
       contents: read
       packages: write
-    if: github.ref == 'refs/heads/develop'
+    #if: github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/groupStrat'
```
Comment (RohitKandimalla) [inline .github/workflows/maven-ci.yml:24]: Is the condition not required?

File: pom.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
   <groupId>gov.cms.madie</groupId>
   <artifactId>madie-java-models</artifactId>
-  <version>0.1.2-SNAPSHOT</version>
+  <version>0.1.2.a-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (ethankaplan) [reply pom.xml:9]: Oh yeah, that was for local testing when dev was broken. 
...I dunno what to do in this case

File: .github/workflows/maven-ci.yml (hunk 1 above)
Comment (gregory-akins) [reply .github/workflows/maven-ci.yml:24]: Seems like we added refs/heads/groupStrat to force a publish to Maven repo for the branch

the original regs/heads/develop caused a merge to develop to publish

We can install locally for testing... but refs/heads/develop should be left intact and not commented out

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gregory-akins) [reply pom.xml:9]: Our versioning strategy sucks on this.  The final commit should be 0.1.2-SNAPSHOT, unless you need to make sure that existing consumers don't use your new version

**Comment by gregory-akins** [comment]
Make sure you update the PR Tempate with the correct JIRA and the appropriate checkboxes checked

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nmorasb) [reply pom.xml:9]: Personally I think we should at least be incrementing the patch number anytime we add new fields. Not changing the version number is what broken local development last Thursday as well as the builds for the test environment, as the models change was merged and published without the corresponding changes in the backend service.
If entirely new models are being added, maybe we don't need to change the version, but adding new fields on existing models changes the constructor signatures and breaks backend code until supporting changes are merged.
And true breaking changes should be a minor version update at minimum.

[the remark this answers is not in the document]
Comment (ethankaplan) [reply pom.xml:9]: So should it be 0.1.3 or 0.1.21 (or something)?

[the remark this answers is not in the document]
Comment (RohitKandimalla) [reply pom.xml:9]: Agree, updating version numbers make sense. I believe we can go with 0.1.3 ( increasing the patch number ).

[the remark this answers is not in the document]
Comment (gregory-akins) [reply pom.xml:9]: That's fine.  We'll codify better release processes for the libraries with the next UAT

[the remark this answers is not in the document]
Comment (ethankaplan) [reply pom.xml]: Updated in new push

File: .github/workflows/maven-ci.yml (hunk 1 above)
Comment (ethankaplan) [reply .github/workflows/maven-ci.yml]: Updated in new push

**Review by sb-cecilialiu — APPROVED** [review]
